src/apps/loan-list/materials/stackable-material/stackable-material.tsx

```diff
@@ -38,16 +38,22 @@ const StackableMaterial: FC<StackableMaterialProps & MaterialProps> = ({
     }
   };
 
+  const handleOpenDueDateModal = () => {
+    if (openDueDateModal && dueDate) {
+      openDueDateModal(dueDate);
+    }
+  };
+
   return (
     <div
       className={clsx("list-reservation my-32 cursor-pointer", {
         "list-reservation--stacked": additionalMaterials > 0
       })}
       role="button"
-      onClick={() => openLoanDetailsModalHandler()}
+      onClick={() => handleOpenDueDateModal()}
```

Comment:
Please call the function directly, as you are creating a new anonymous arrow function on each render. 
`onClick={openLoanDetailsModalHandler}` is preferable in this situation